### PR TITLE
[uikit] Add some `advice` to UISearchController's constructors

### LINKED ISSUE #4114: Null UISearchBar when creating a new UISearchController without arguments
Hi! Just encountered an [old bug](https://bugzilla.xamarin.com/show_bug.cgi?id=37953) that still seems to be present: creating a `UISearchController` without arguments leads to a missing `UISearchBar` inside it.

### Steps to Reproduce

`var searchController = new UISearchController();`

### Actual Behavior

`searchController.SearchBar` is `null`

Creating the controller with `new UISearchController((UIViewController)null)` fixes the problem.

### Environment

- iOS 11.3
- VS4Mac 7.5.1
- Mono 5.10.1.47

### Example Project (If Possible)

[SearchControllerBug.zip](https://github.com/xamarin/xamarin-macios/files/2028896/SearchControllerBug.zip)

Thank you!

### PR DESCRIPTION
**Pull request by joconte**
- Fixes #4114

### CODE REVIEW

**Comment by dalexsoto** [comment]
build

**Comment by dalexsoto** [comment]
Thank you so much for your contribution! ❤️

**Comment by joconte** [comment]
You're welcome ! And thanks to the whole team for helping me contributing ❤️

File: src/uikit.cs [hunk 1]
```diff
@@ -10220,12 +10220,21 @@ interface UISearchContainerViewController
...
+		[Advice("It's recommended to use the constructor that takes a UIViewController searchResultsController in order to create/initialize an attached UISearchBar.")]
+		IntPtr Constructor ();
+
 		[Export ("initWithNibName:bundle:")]
 		[PostGet ("NibBundle")]
 		IntPtr Constructor ([NullAllowed] string nibName, [NullAllowed] NSBundle bundle);
 
 		[Export ("initWithSearchResultsController:")]
+		[DesignatedInitializer]
+		[Advice("You can pass a null UIViewController to display the search results in the same view.")]
```
Comment (chamons) [inline src/uikit.cs]: Nitpick - spacing
```suggestion
		[Advice ("You can pass a null UIViewController to display the search results in the same view.")]
```

File: src/uikit.cs [hunk 2]
```diff
@@ -10220,12 +10220,21 @@ interface UISearchContainerViewController
 	
 	[iOS (8,0)]
 	[BaseType (typeof (UIViewController))]
-	partial interface UISearchController : UIViewControllerTransitioningDelegate, UIViewControllerAnimatedTransitioning {
+	[DisableDefaultCtor] // designated
+	partial interface UISearchController : UIViewControllerTransitioningDelegate, UIViewControllerAnimatedTransitioning
+	{
+		[Export ("init")]
+		[DesignatedInitializer]
+		[Advice("It's recommended to use the constructor that takes a UIViewController searchResultsController in order to create/initialize an attached UISearchBar.")]
```
Comment (chamons) [inline src/uikit.cs]: Nitpick - spacing
```suggestion
		[Advice ("It's recommended to use the constructor that takes a UIViewController searchResultsController in order to create/initialize an attached UISearchBar.")]
```

**Comment by dalexsoto** [comment]
build

**Comment by rolfbjarne** [comment]
Thanks a lot!

The test failure is unrelated (https://github.com/xamarin/maccore/issues/1742).

**Comment by VincentDondain** [comment]
@joconte FYI I updated your PR's title and and description to match our guidelines and make use of https://help.github.com/en/articles/closing-issues-using-keywords.

For future reference we usually add in "[]" the area of the product or framework you changed (in this case UIKit) and then a verbal description of the change. Then in the description if you do "fixes #XXX or closes #XXX" (check the GitHub link for more info) it will automatically close the issue when we merge the PR. Otherwise if your syntax is wrong you'll have to remember to close the issue manually (:

**Comment by VincentDondain** [comment]
And @rolfbjarne the test failure isn't quite unrelated:

`!extra-designated-initializer! UISearchController::initWithSearchResultsController: is incorrectly decorated with an [DesignatedInitializer] attribute`

Only the `GameConroller` xtro report is unrelated.

File: src/uikit.cs [hunk 3]
```diff
@@ -10220,12 +10220,21 @@ interface UISearchContainerViewController
 	
 	[iOS (8,0)]
 	[BaseType (typeof (UIViewController))]
-	partial interface UISearchController : UIViewControllerTransitioningDelegate, UIViewControllerAnimatedTransitioning {
+	[DisableDefaultCtor] // designated
+	partial interface UISearchController : UIViewControllerTransitioningDelegate, UIViewControllerAnimatedTransitioning
+	{
+		[Export ("init")]
+		[DesignatedInitializer]
```
Comment (VincentDondain) [inline src/uikit.cs]: I think this one shouldn't be `[DesignatedInitializer]`
```suggestion

```

According to Apple's header files:

```
Headers/UISearchController.h
39:- (instancetype)initWithSearchResultsController:(nullable UIViewController *)searchResultsController NS_DESIGNATED_INITIALIZER;
```

Comment (spouliot) [reply src/uikit.cs]: Agreed. It's not in the headers nor in the documentation
https://developer.apple.com/documentation/uikit/uisearchcontroller?language=objc

**Comment by VincentDondain** [comment]
build

**Comment by VincentDondain** [comment]
Mmh xtro is still not happy:

```
!extra-designated-initializer! UISearchController::initWithSearchResultsController: is incorrectly decorated with an [DesignatedInitializer] attribute
```

That's weird cause the headers definitely say it's a `NS_DESIGNATED_INITIALIZER`.

**Comment by VincentDondain** [comment]
Aaaaaaah I know why 🤦‍♂ 

We're ahead of our time here, it's Xcode 11 that made `UISearchController::initWithSearchResultsController:` a `NS_DESIGNATED_INITIALIZER`. This PR is against master, so Xcode 10.2 and we shouldn't have the designated initializer.

File: src/uikit.cs [hunk 4]
```diff
@@ -10220,12 +10220,20 @@ interface UISearchContainerViewController
...
+		[Export ("init")]
+		[Advice ("It's recommended to use the constructor that takes a 'UIViewController searchResultsController' in order to create/initialize an attached 'UISearchBar'.")]
+		IntPtr Constructor ();
+
 		[Export ("initWithNibName:bundle:")]
 		[PostGet ("NibBundle")]
 		IntPtr Constructor ([NullAllowed] string nibName, [NullAllowed] NSBundle bundle);
 
 		[Export ("initWithSearchResultsController:")]
+		[DesignatedInitializer]
```
Comment (VincentDondain) [inline src/uikit.cs]:
```suggestion

```

Please just remove that. I can't push to your fork cause you're using a master branch. In the future please create a branch (even on your fork).

Comment (joconte) [reply src/uikit.cs]: Thanks for the tips.

**Comment by VincentDondain** [comment]
build

**Pull request merged**